src/main/java/seedu/address/model/wedding/WeddingDate.java

```diff
@@ -22,6 +23,20 @@ public WeddingDate(LocalDate date) {
         fullDate = date;
     }
 
+    /**
+     * Returns true if a given string is a valid wedding date.
+     */
+    public static boolean isValidWeddingDate(String dateString) {
+        requireNonNull(dateString);
+        try {
```

Comment:
Regarding your concern that using try-catch for control flow might be an issue, I think it should be fine because it does seem unavoidable in this case since localDate will throw an exception, and it's quite cumbersome to create a regex pattern to check for proper pattern matching for dates (would have to handle leap years/different numbers etc.)